snap/quota/quota.go

```diff
@@ -200,7 +200,17 @@ func (grp *Group) NewSubGroup(name string, memLimit quantity.Size) (*Group, erro
 func ResolveCrossReferences(grps map[string]*Group) error {
 	// iterate over all groups, looking for sub-groups which need to be threaded
 	// together with their respective parent groups from the set
-	for name, grp := range grps {
+
+	// iterate in the same order
+	grpNames := make([]string, 0, len(grps))
+	for name := range grps {
+		grpNames = append(grpNames, name)
+	}
+
+	sort.Strings(grpNames)
```

Comment:
I forgot about this PR and did this initially for my fix in https://github.com/snapcore/snapd/pull/10206 as well but then felt it's overkill. If this doesn't return errors the order doesn't matter, or is there something else influenced by that? if it does then returning a multi error would seem better.

Reply:
this has been removed